### ISSUE TITLE
Invisible Submission (or something to that effect)

### ISSUE THREAD

**Issue by mello99** [VTUL/vtechworks#155]
Hi all, we recently received a submission from Dr. Xin Luo entitled "Paradoxical effects of all-trans-retinoic acid on ..." to the "Research Articles, (VMRCVM)" (http://vtechworks.lib.vt.edu/handle/10919/23146) collection. I took the task to see if I could approve the item, but when I clicked on the item, I was taken to a completely blank metadata record (no file or anything).

<img width="755" alt="invisible_submission" src="https://cloud.githubusercontent.com/assets/5408616/12131067/694ca658-b3dd-11e5-889c-9664798473ba.png">

I then clicked on "Full Item Record" and was taken to a similar page: 

<img width="736" alt="full_item_record_invisible" src="https://cloud.githubusercontent.com/assets/5408616/12131082/8c975f9a-b3dd-11e5-8c84-92d37f47dab1.png">

Could this be a problem on our end, @alawvt and @keithgee , or does it involve something the user did? I always thought that the user had to provide at least 1) a title, 2) a date, and 3) a file in order to complete a VTW submission, but I could be wrong. Based on the name of the submission it looks like the user provided a title, but I'm lost as to what happened thereafter. If the issue is not on our end, I'll contact the user and ask him or her to resubmit (I'll contact the user regardless, but I wanted to figure out what was happening first). Thanks so much, and take care.

**Comment by mello99** [comment]
Okay, so I clicked on "Edit Metadata" and saw the actual metadata associated with the file; it must be an interface thing with the new version of DSpace. I'll get to this in the morning.

**Comment by alawvt** [comment]
Hmmm. I submitted a test item to this collection and it looked normal, see below. All the metadata that I submitted appears.

![itemsubmission20160106](https://cloud.githubusercontent.com/assets/13037168/12143552/8606600a-b44f-11e5-9c03-8e0c8abb7eea.png)

**Comment by amandafrench** [comment]
Maybe try one more test, @alawvt and @mello99, where one of you submits and item and the other tries to claim the task? If we can't reproduce the issue, though, let's close it. If it happens again we can reopen it.

**Comment by mello99** [comment]
@alawvt and I tried it again. Anne submitted an item herself and was able to see the full metadata record (see above). 

I still wasn't able to see the full metadata record for the item in question (nor was Anne), but I was nonetheless able to approve the item and correct/add to the existing metadata. The submitter provided very little metadata altogether; the submitter also mistakenly designated the item as being in the public domain (I need to keep track of how often this happens). The item is now publicly available on DSpace - https://vtechworks.lib.vt.edu/handle/10919/64413. I'll go ahead and close this issue since it appears to be a one-time glitch; if it happens again I'll re-open it, like you suggested.

**Comment by alawvt** [comment]
I just had the same experience. I received notice of an OMALS submission and the screen was blank.
![nometadatasubmission](https://cloud.githubusercontent.com/assets/13037168/12650827/550106bc-c5b2-11e5-820e-c6ae8e36902a.png)

I could see the metadata by clicking on "Edit metadata."
![nometadatasubmissionfilemetadata](https://cloud.githubusercontent.com/assets/13037168/12651148/6b50ed0a-c5b3-11e5-9500-b1a026f19272.png)

When I clicked on the file name it produced an error,
![nometadatasubmissionmissingfile](https://cloud.githubusercontent.com/assets/13037168/12651189/8be3a346-c5b3-11e5-9257-1cc25fa95d83.png)

**Comment by alawvt** [comment]
I approved the submission and it appears in the collection with file description, which is ok. 

![nometadatasubmissionafterupload](https://cloud.githubusercontent.com/assets/13037168/12651530/f0bfa11a-c5b4-11e5-9d7d-2539c11506ca.png)

When I click on the file, there is another error.
![nometadatasubmissionafteruploaderror](https://cloud.githubusercontent.com/assets/13037168/12651566/09fc3904-c5b5-11e5-915e-936437b526c5.png)

I will write to the author and request the file.

**Comment by AmberPoo1** [comment]
Found the similar problem when doing the "Test latest changes on vtechworks.vt.edu since Dec. 1. ". I created an test item, no content shows when click on the "show full item record" or "show simple item record".
![screen shot 2016-01-29 at 11 09 39 am](https://cloud.githubusercontent.com/assets/13734326/12680659/0c7c99c0-c679-11e5-8431-12950b1fe985.png)

**Comment by alawvt** [comment]
The head of OMALS has reported this same issue with multiple submissions to that collection, http://vtechworks.lib.vt.edu/handle/10919/51287.

I reported this issue as "Metadata missing in Item Approval View," https://jira.duraspace.org/browse/DS-3216 on 2016-05-20.

**Comment by amandafrench** [comment]
If (when) this occurs again, I'd suggest clicking on the "show" links on the error page and giving the full error messages to @keithgee and/or ITS, perhaps in a comment on this issue. Could be a system error instead of an application error. Will close for now since there's probably not much we can do about it now.

**Comment by alawvt** [comment]
Since ITS changed the error page, we won't be able to click on a link to the stack trace. It is probably recorded in the log, but I don't know how well or hard it would be to find. The error page only shows up when clicking on the file link. The other screens weren't error messages.

**Comment by keithgee** [comment]
@alawvt which specific items have had this problem recently?  I notice that we are now keeping only about 3 months of log files on the server, so it is more difficult to investigate the items from January.

**Comment by alawvt** [comment]
From the latest report, I would imagine at least two of the items in the OMALS collection submitted in 2016 had this problem.

**Comment by alawvt** [comment]
Just had another case loading http://hdl.handle.net/10919/71289 (5/27/2016 ~3pm). Here are screenshots:
![blanckapproval_1](https://cloud.githubusercontent.com/assets/13037168/15618686/7ea4b5e4-241d-11e6-9b8b-addeee2fe6af.png)

![blanckapproval_2](https://cloud.githubusercontent.com/assets/13037168/15618701/8d377312-241d-11e6-9c34-37ce15d533ab.png)

**Comment by keithgee** [comment]
I'm not sure where to go with this yet; the log files don't show anything unusual. I may try to replicate on my LDE when I get a chance.

@mello99 and @alawvt , if this is seen again would it be okay to not accept or edit the item until I can look at it? I wonder if I could spot any differences that would show why no information appears.

**Comment by mello99** [comment]
Sure, @keithgee , I'll let you know if we encounter those issues again and I won't approve anything until I get the go-ahead from you. Take care.

**Comment by mello99** [comment]
Hi @keithgee , I ran into another invisible submission when Anita submitted her latest paper after 6pm today. I've returned it to the task pool so that you can look at it. Thanks.

**Comment by keithgee** [comment]
@mello99, could I borrow your username and password sometime?(but don't post it here). I don't have access to the task pool for that collection, and adding my account to that list of people now still wouldn't let me see it.

Thanks!

**Comment by mello99** [comment]
Hi @keithgee , I created 3 test submissions thus far:

One of them, "Test invisible submission 3”, is not showing any of the metadata (the submission is in the task pool and you should be able to look at it and take it when logged-in to VTechWorks). 

The other two submissions showed the metadata associated with the records – these two records were submitted to the “Library Test” collection, whereas the invisible submission item was submitted to the “VTechWorks Documentation” collection. Some other distinctions are that “Test invisible submission 3” had a Creative Commons license attached to it and was classified as a presentation, whereas the others did not have Creative Commons licenses attached and were classified as "Other" (I think). I’ll keep testing and see what I find. Thanks, take care.

**Comment by mello99** [comment]
@keithgee , also created another submission, "Test Invisible Submission 4" - that doesn't show the item's record, either.

**Comment by keithgee** [comment]
@mello99 thanks for creating these. I haven't had time to sort it out yet, [I only get a few minutes here and there], but it appears to be a problem with the internal mets.xml file that DSpace generates.

We had a previous issue with "blank items" that were already accepted, and in that particular problem it turned out that some groups had been deleted in VTechWorks (I'll try to find a reference somewhere).

I don't think it's quite the same thing - I don't see how that would apply - but I do think it's something that's going on with the mets.xml.

**Comment by keithgee** [comment]
Here is a link to the issue that I referenced earlier, about problems with mets.xml because of deleted groups. Probably not the same thing, but this may help remind me how I pinpointed the issue:
https://github.com/VTUL/vtechworks/issues/169

**Comment by keithgee** [comment]
I have made a fix for this problem with commit  3e5f329 and issued a pull request.
We have some custom code that displays the lock for campus-only items, even when the user is on campus, that was added with commit 9b6d962.
The code had a problem that caused VTechWorks to not be able to get the metadata for items under certain conditions, which made the display look blank for some submissions in the workflow.
I suspect this fix may also correct other problems where item pages appeared blank.

@alawvt you might want to look closely at the code in the pull request before merging, as I'm not entirely sure about the intent of the code added in the original commit. Perhaps @pgray64 and @mello99 might help test.

@amandafrench this might be a candidate to test hot-fix procedures, although we've lived with this problem for a while.
